castledio/castled
# persisting onboarding context across lifecycle

Frontend 
- Onboarding context across application lifecycle
- Handling edge cases for context and onboarding count

Backend
- Bug fix on model delete for 0 pipelines associated.

### QA Checklist Report:


**Test Name | Tested on OSS?(Yes/No) | Tested on Enterprise?(Yes/No)**
-- | -- | --
Clear DB:DROP DATABASE DB_NAME;CREATE DATABASE DB_NAME; | Yes | Yes
**Login** |   |  
Can you sign-in? (IS_OSS=true) in .env | Yes | Not Applicable
Can you sign in? (IS_OSS=false) in .env | Not Applicable | Yes
Can you log out? (IS_OSS=false) in .env | Yes | Yes
**Welcome** |   |  
Can you see two buttons?(Create using Demo & Own warehouse) | Yes | Yes
Can you see the “steps left” on each button? | Yes | Yes
Does clicking on both buttons open the onboarding steps dropdown? (For demo 3rd step, For own 1st step will open) | Yes | Yes
Does clicking on buttons in the dropdown, redirects us to respective screens)E.g. "+ Add App" should redirect to the "Create Destination" page. | Yes | Yes
Can you go back using the "Go Back" button | Yes | Yes
Does clicking on one dropdown close the other? | Yes | Yes
**Warehouse** |   |  
Click on the "Warehouses" button on the sidebar, open warehouse list page? | Yes | Yes
Can you see the Demo warehouse in Enterprise? | Yes | Yes
Clicking the "Add Warehouse" button opens the "Select Source Type" section with a tabbed list of all warehouses? | Yes | Yes
Can you successfully create a warehouse? | Yes | Yes
When clicking on a warehouse from /warehouses page, opens the warehouse config page? | Yes | Yes
Can you see the banner? (If creating a pipeline for the first time) | Yes | Yes
After creation, Go to /welcome page and click on the "Own warehouse" button. Does 1st section come auto checked and 2nd section auto opened? | Yes | Yes
**Model** |   |  
Click on the "Models" button on the sidebar, open the model list page? | Yes | Yes
Clicking the "Create Model" button opens the "Select Source Type" section with a tabbed list of all warehouses? | Yes | Yes
Can you see demo_model on enterprise? | Yes | Yes
Can you successfully create a model with an existing warehouse? | Yes | Yes
When clicking on a model from /models page, opens the model details page? | Yes | Yes
Can you see three tabbed lists on the model details' page namely "Query", "Column Details", "Pipelines" with respective details on each section? | Yes | Yes
Clicking on the "Create Pipeline" button on the model details page, redirects to the Run Model page with model query prefilled? | Yes | Yes
Can you "Delete Model"? (If no pipelines are created) | Yes | Yes
Can you see the banner? (If creating a pipeline for the first time) | Yes | Yes
After creation, Go to /welcome page and click on the "Own warehouse" button. Does the 2nd section come auto checked and the 3rd section auto opened? | Yes | Yes
**Apps** |   |  
Click on the "Apps" button on the sidebar to open the Apps list page? | Yes | Yes
Clicking the "Add App" button opens the "Select Destination Type" section with a tabbed list of all destinations? | Yes | Yes
Can you successfully add an app? | Yes | Yes
When clicking on an app from /apps page, opens the app config page? | Yes | Yes
Can you see the banner? (If creating a pipeline for the first time) | Yes | Yes
After creation, Go to /welcome page and click on the "Own warehouse" button. Does the 3rd section come auto checked and the 4th section auto opened? | Yes | Yes
**Pipeline** |   |  
On 3 - Mapping page, You should not encounter any Error with details not matching the field selected?E.g. "App fields repeating" error even if app fields are not repeating, and so on. | Yes | Yes
Can you delete rows? (If any) | Yes | Yes
Does it give a mandatory error if no mandatory is selected?(If already selected, delete the values from localStorage and refresh the page, and then try “Test & continue”. | Yes | Yes
After clicking on test and continue and coming back to the mapping page, are the fields getting pre-populated with the values you've selected earlier? | Yes | Yes
Click on test and continue and create a pipeline successfully? | Yes | Yes
Are you redirected to the pipeline details page, for which you have created the pipeline? | Yes | Yes
Can you see three tabbed lists on the model detail's page namely "Runs", "Query & Mapping", "Schedule" with respective details on each section? | Yes | Yes
After creation, Go to /welcome page, does it redirect to /pipelines page? | Yes | Yes
Banner is hidden now? | Yes | Yes
When clicking on a pipeline from /pipelines page, opens the pipeline details page? | Yes | Yes
Can you delete Pipeline? | Yes | Yes
After deletion, are you redirected to /welcome page, with both buttons saying, 1 step left? | Yes | Yes
Can you create pipeline using demo warehouse? | Yes | Yes
Can you see pipeline details on model details section? | Yes | Yes
**Deployment** |   |  
Does yarn && yarn build works locally? | Yes | Yes
